plugins/woocommerce/includes/export/class-wc-product-csv-exporter.php

```diff
@@ -714,12 +714,7 @@ protected function prepare_meta_for_export( $product, &$row ) {
 						continue;
 					}
 
-					// Allow 3rd parties to process the meta, e.g. to transform non-scalar values to scalar.
-					$meta_value = apply_filters( 'woocommerce_product_export_meta_value', $meta->value, $meta, $product, $row );
-
-					if ( ! is_scalar( $meta_value ) ) {
-						continue;
-					}
+					$meta_value = apply_filters( 'woocommerce_product_export_meta_value', maybe_serialize( $meta->value ), $meta, $product, $row );
```

Comment:
I think this makes the change somewhat backward incompatible. Extensions that previously got a raw value (and potentially serialized it) will now get a serialized value. I think the better option here would be to run the serialization only after the filter is applied. What do you think?

That said, this can easily lead to double serialization. We can't really avoid doing that, as during the import, we wouldn't know whether to unserialize or not...